sdk/plugin/v5/plugin.go

```diff
@@ -0,0 +1,176 @@
+package plugin
```

Comment:
I'm wondering if we really need this to be in its own package? It's unfortunate in my opinion that we now have to export more types and variables to support that. Since this is in the SDK i want to be careful about exporting items that we don't actually want people to import and use.

Reply:
We had a discussion about this and we decided to move this code into the existing package so that we don't have to export fields in the sdk that should be kept private.